core-bundle/src/DataContainer/DcaUrlAnalyzer.php

```diff
@@ -43,19 +43,25 @@ public function __construct(
     /**
      * @return array{string|null, int|null}
      */
-    public function getCurrentTableId(): array
+    public function getCurrentTableId(Request|string|null $request = null): array
     {
-        $this->request = $this->requestStack->getCurrentRequest() ?? throw new \LogicException('Unable to retrieve DCA information from empty request stack.');
+        $request ??= $this->requestStack->getCurrentRequest() ?? throw new \LogicException('Unable to retrieve DCA information from empty request stack.');
+
+        if (!$request instanceof Request) {
+            $request = Request::create($request);
+        }
```

Comment:
This can never be true because of the exception above, can it? `Request::getCurrentRequest()` returns either `null` or a `Request` object.